docs/doc/60-contributing/03-rfcs/20221017-multiple-catalog.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Multiple Catalog
+description: multiple catalog support for databend
+---
+
+- RFC PR: [datafuselabs/databend#8255](https://github.com/datafuselabs/databend/pull/8255)
+- Tracking Issue: [datafuselabs/databend#8300](https://github.com/datafuselabs/databend/issues/8300)
+
+## Summary
+
+Allow users to maintain multiple catalogs for the databend.
+
+## Motivation
+
+Databend organize data in three layers:
+
+```txt
+catalog -> database -> table
+```
+
+- `catalog`: the biggest layer for databend, contains all databases and tables, provided by [`Catalog`](https://github.com/datafuselabs/databend/blob/556aedc00e5e8a95a7551d0ec21b8e6fa7573e0a/src/query/catalog/src/catalog.rs#L80)
+- `database`: the container of tables, provided by [`Database`](https://github.com/datafuselabs/databend/blob/556aedc00e5e8a95a7551d0ec21b8e6fa7573e0a/src/query/catalog/src/database.rs#L44)
+- `table`: the smallest unit in databend, provided by [`Table`](https://github.com/datafuselabs/databend/blob/556aedc00e5e8a95a7551d0ec21b8e6fa7573e0a/src/query/catalog/src/table.rs#L44)
+
+By default, all databases and tables will be stored in `default` catalog (powered by `metasrv`).
+
+Databend supports multiple catalogs now, but only in a static way.
+
+To allow accessing the `hive` catalog, users need to configure `hive` inside `databend-query.toml` in this way:
+
+```toml
+[catalog]
+meta_store_address = "127.0.0.1:9083"
+protocol = "binary"
+```
+
+Users can't add/alter/remove the catalogs during runtime.
+
+By allowing users to maintain multiple catalogs for the databend, we can integrate more catalogs like `iceberg` more quickly.
+
+## Guide-level explanation
+
+After this RFC has been implemented, users can create new catalogs like:
+
+```sql
+CREATE CATALOG my_hive
+  TYPE='hive'
+  CONNECTION = (URL='<hive-meta-store>' THRIFT_PROTOCOL=BINARY);
+SELECT * FROM my_hive.DB.table;
+```
+
+Besides, users can alter or drop a catalog:
+
+```sql
+DROP CATALOG my_hive;
+```
+
+Users can add more catalogs like:
+
+```sql
+CREATE CATALOG my_iceberg
+  TYPE='iceberg'
+  CONNECTION = (URL='s3://my_bucket/path/to/iceberg');
```

Comment:
```suggestion
CREATE CATALOG my_iceberg
  TYPE='iceberg'
  LOCATION = (
    URL='s3://my_bucket/path/to/iceberg'
    CONNECTION = (
      ACCESS_KEY_ID = ...
      ...
    )
  )
```

Let's create catalogs like creating stages.

Reply:
Not all catalog types will have concepts like `URL`. I prefer to leave them for catalog providers to decide.